Sources/SwiftGen/Logs.swift

```diff
@@ -39,17 +39,37 @@ enum ANSIColor: UInt8, CustomStringConvertible {
   }
 }
 
+// Based on https://github.com/realm/SwiftLint/blob/0.39.2/Source/SwiftLintFramework/Extensions/QueuedPrint.swift
+
 func logMessage(_ level: LogLevel, _ string: CustomStringConvertible) {
-  switch level {
-  case .info:
-    fputs(ANSIColor.green.format("\(string)\n"), stdout)
-  case .warning:
-    fputs(ANSIColor.yellow.format("swiftgen: warning: \(string)\n"), stderr)
-  case .error:
-    fputs(ANSIColor.red.format("swiftgen: error: \(string)\n"), stderr)
+  outputQueue.async {
+    switch level {
+    case .info:
+      fputs(ANSIColor.green.format("\(string)\n"), stdout)
+    case .warning:
+      fputs(ANSIColor.yellow.format("swiftgen: warning: \(string)\n"), stderr)
+    case .error:
+      fputs(ANSIColor.red.format("swiftgen: error: \(string)\n"), stderr)
+    }
   }
 }
 
+private let outputQueue: DispatchQueue = {
+  let queue = DispatchQueue(
+    label: "swiftgen.outputQueue",
+    qos: .userInteractive,
+    target: .global(qos: .userInteractive)
+  )
+
+  #if !os(Linux)
+  atexit_b {
+    queue.sync(flags: .barrier) {}
+  }
```

Comment:
nit: I understand what this does (ensuring every pending log message gets dequeued before we really exit) but given that `atexit_b` is not a common function known to every devs, a code comment could be welcome